shap/explainers/kernel.py

```diff
@@ -370,7 +374,13 @@ def varying_groups(self, x):
         varying = np.zeros(len(self.data.groups))
         for i in range(0, len(self.data.groups)):
             inds = self.data.groups[i]
-            num_mismatches = np.sum(np.abs(x[0, inds] - self.data.data[:, inds]) > 1e-7)
+            x_group = x[0, inds]
+            if sp.sparse.issparse(x_group):
+                if i not in x.nonzero()[1]:
```

Comment:
`i` is a group index not an index into `x` (that is what `inds` is) so while they are the same when there is no grouping this won't work when some features are grouped.

Reply:
Also we need to make sure both the background data and the current x are 0 otherwise the feature does vary.

Reply:
I wonder if in the sparse case there should be some threshold for the background?  If the background is mostly zero but only has a single nonzero, and the feature in the evaluated sample is zero, then the feature importance will be nonzero but also very low, since most of the time we will be choosing zeros from the background.  To optimize execution time maybe we should just ignore those columns too?

Reply:
I think if the user has decided to pass many background values then they have made the choice to spend lots of time comparing to many samples, so we shouldn't short-circuit that when the input matrix happens to be in the sparse format. That being said, I think for sparse data people should mostly just pass a background with one samples of all 0's since that is by far the most common value.

Reply:
great idea, will update the code, thanks!

Reply:
done, changed to use inds instead of i when checking if group varies